sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -55,17 +57,27 @@ export class ClientCertificateCredential implements TokenCredential {
     tenantId: string,
     clientId: string,
     certificatePath: string,
-    options?: TokenCredentialOptions
+    options?: ClientCertificateCredentialOptions
   ) {
     this.identityClient = new IdentityClient(options);
     this.tenantId = tenantId;
     this.clientId = clientId;
     this.certificateString = readFileSync(certificatePath, "utf8");
 
-    const certificatePattern = /(-+BEGIN CERTIFICATE-+)(\n\r?|\r\n?)([A-Za-z0-9+/\n\r]+=*)(\n\r?|\r\n?)(-+END CERTIFICATE-+)/;
-    const matchCert = this.certificateString.match(certificatePattern);
-    const publicKey = matchCert ? matchCert[3] : "";
-    if (!publicKey) {
+    const certificatePattern = /(-+BEGIN CERTIFICATE-+)(\n\r?|\r\n?)([A-Za-z0-9+/\n\r]+=*)(\n\r?|\r\n?)(-+END CERTIFICATE-+)/g;
```

Comment:
Does this capture headers, footers, and line breaks? x5c should contain only the cert content. For example, given
```
-----BEGIN RSA PRIVATE KEY-----
...
-----END RSA PRIVATE KEY-----
-----BEGIN CERTIFICATE-----
foo
bar
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
baz
-----END CERTIFICATE-----
```
the x5c value would be
```
["foobar","baz"]
```

Reply:
It matches multiple parts, but we only build the array from the certificate. See line 76 below.